### PR TITLE
Fix Datapart functions to be Timezone Invariant

### PR DESCRIPTION
**Pull request by shah-nirmit**
### Description
We were incorrectly passing timezone to `timestamp2tm` function which would cause the datepart functions to be timezone dependent , If we pass tzp by reference to timestamp2tm function, then internally localtime converts the given original timestamp value to date and time values adding timezone offset taking GMT as a reference timezone. However, if NULL is passed instead then we keep the original timestamp intact and that's what actually PG does.

### Issues Resolved

BABEL-5453

Signed-off-by: Nirmit Shah [nirmisha@amazon.com](mailto:nirmisha@amazon.com)

### Test Scenarios Covered ###
* **Use case based -** YES

* **Boundary conditions -** YES

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/input/functions/datepart-vu-prepare.sql [hunk 1]
```diff
@@ -43,3 +43,39 @@ CREATE PROCEDURE date_part_vu_prepare_sys_day_proc @a datetime
 AS
 SELECT DAY(@a)
 GO
+
+-- Test Case for Date Part Functions Timezone Invariance
+CREATE TABLE date_part_vu_prepare_TestDates (
+    DateID INT IDENTITY(1,1) PRIMARY KEY,
+    TestDate DATETIME
```
Comment (Anikait143) [inline test/JDBC/input/functions/datepart-vu-prepare.sql]: We should also test for other datatypes as well - `DATETIMEOFFSET`, `DATETIME2`, `SMALLDATETIME`

File: test/JDBC/input/functions/datepart-vu-verify.sql [hunk 2]
```diff
@@ -210,3 +210,54 @@ GO
...
+
+    -- Insert results for this timezone
+    INSERT INTO date_part_vu_prepare_TestResults (TestCase, TimeZone, InputDate, Day, Month, Year)
+    SELECT 
+        'Test: ' + CONVERT(VARCHAR, d.TestDate, 120) + ' in ' + @current_timezone AS TestCase,
+        @current_timezone AS TimeZone,
+        d.TestDate AS InputDate,
+        DAY(d.TestDate) AS [Day],
+        MONTH(d.TestDate) AS [Month],
+        YEAR(d.TestDate) AS [Year]
```
Comment (Anikait143) [inline test/JDBC/input/functions/datepart-vu-verify.sql]: Please also add tests for `datepart` and `datename` components listed in TSQL docs.

**Review by sumitj824 — APPROVED** [review]

**Review by Anikait143 — APPROVED** [review]

**Review by jsudrik — APPROVED** [review]
Approved.
